backend.py

```diff
@@ -1280,6 +1281,36 @@ def writeFstab(mounts, primary_disk, logs_partnum, swap_partnum, disk_label_suff
     if logs_partition:
         fstab.write("LABEL=%s    /var/log         %s     defaults   0  2\n" % (logsfs_label%disk_label_suffix, fs_type))
 
+def configureSSHMode(mounts, ssh_mode):
+    """Configure SSH mode according to answerfile settings."""
+
+    if ssh_mode is None or ssh_mode not in ['auto', 'on', 'off']:
+        return
+
+    logger.log("Configuring SSH mode: %s" % ssh_mode)
+
+    ssh_auto_mode_conf_path = os.path.join(mounts['root'], 'etc/xapi.conf.d/ssh-auto-mode.conf.extra')
+    conf_dir = os.path.dirname(ssh_auto_mode_conf_path)
+    util.assertDir(conf_dir)
+
+    if ssh_mode == 'auto':
+        with open(ssh_auto_mode_conf_path, 'w') as f:
+            f.write('ssh-auto-mode=true\n')
+        logger.log("SSH auto-mode enabled")
+    else:
+        with open(ssh_auto_mode_conf_path, 'w') as f:
+            f.write('ssh-auto-mode=false\n')
+
+        # Enable or disable SSH service based on mode
+        if ssh_mode == 'on':
+            util.runCmd2(['chroot', mounts['root'], 'systemctl', 'enable', 'sshd.service'])
+            util.runCmd2(['chroot', mounts['root'], 'systemctl', 'start', 'sshd.service'])
```

Comment:
I don't think starting or stopping the service makes sense in the install environment.

Reply:
Yes, this could be a problem, beside the mentioned conflict between installer ssh and system ssh launching a daemon could cause issue when unmounting file systems. Probably as systemd is running outside the chroot this is probably going to fail anyway.